app/lib/katello/resources/candlepin.rb

```diff
@@ -564,40 +554,11 @@ def path(id = nil)
       end
 
       class Pool < CandlepinResource
-        class << self
-          def find(pool_id)
-            pool_json = self.get(path(pool_id), self.default_headers).body
-            fail ArgumentError, "pool id cannot contain ?" if pool_id["?"]
-            JSON.parse(pool_json).with_indifferent_access
-          end
-
-          def get_for_owner(owner_key, include_temporary_guests = false)
-            url = "/candlepin/owners/#{owner_key}/pools?add_future=true"
-            url += "&attribute=unmapped_guests_only:!true" if include_temporary_guests
-            pools_json = self.get(url, self.default_headers).body
-            JSON.parse(pools_json)
-          end
-
-          def destroy(id)
-            fail ArgumentError, "pool id has to be specified" unless id
-            self.delete(path(id), self.default_headers).code.to_i
-          end
-
-          def entitlements(pool_id, included = [])
-            entitlement_json = self.get("#{path(pool_id)}/entitlements?#{included_list(included)}", self.default_headers).body
-            JSON.parse(entitlement_json)
-          end
+        prepend PoolRestClient
+      end
 
-          def path(id = nil, owner_label = nil)
-            if owner_label && id
-              "/candlepin/owners/#{owner_label}/pools/#{id}"
-            elsif owner_label
-              "/candlepin/owners/#{owner_label}/pools/"
-            else
-              "/candlepin/pools/#{id}"
-            end
-          end
-        end
+      class UpstreamPool < UpstreamCandlepinResource
```

Comment:
What do you think about putting things like UpstreamPool into a module? => Katello::Resources::Upstream::Whatever so we can break this large file apart a bit more

Reply:
Yes, and split all these classes into separate files under `resources/candlepin/` and have `resource/candlepin.rb` be base + `require`s, but I was going to do that (latter) step in a separate commit.